docs/odh/oci-model-storage.md

```diff
@@ -8,15 +8,228 @@ which also explains how to deploy models from OCI images.
+  --build-arg repo_id=ibm-granite/granite-3.0-2b-instruct \
+  -t quay.io/<user_name>/<repository_name>:<tag_name> .
+
+podman push quay.io/<user_name>/<repository_name>:<tag_name>
+```
+
+It is important to use the `--squash` flag to prevent the final image having
+the double size of the model.
+
+> [!TIP]
+> When uploading your container image, if your repository is private, ensure you
+> are authenticated to the registry.
+
+### Deploying the Granite model using the generated OCI image
+
+Start by creating a namespace to deploy the model:
+```shell
+oc new-project oci-model-example
+```
+
+In the newly created namespace, you need to create a `ServingRuntime` resource
+configuring vLLM model server. The ODH project provides templates with
+configurations for some model servers, which you can list with the following
+command:
+```shell
+oc get template -n opendatahub
+
+NAME                                 DESCRIPTION                                                                        PARAMETERS    OBJECTS
+caikit-standalone-serving-template   Caikit is an AI toolkit that enables users to manage models through a set of...    0 (all set)   1
+caikit-tgis-serving-template         Caikit is an AI toolkit that enables users to manage models through a set of...    0 (all set)   1
+kserve-ovms                          OpenVino Model Serving Definition                                                  0 (all set)   1
+ovms                                 OpenVino Model Serving Definition                                                  0 (all set)   1
+tgis-grpc-serving-template           Text Generation Inference Server (TGIS) is a high performance inference engin...   0 (all set)   1
+vllm-runtime-template                vLLM is a high-throughput and memory-efficient inference and serving engine f...   0 (all set)   1
+```
+
+The template that is applicable for KServe and holds the vLLM configuration
+is the one named as `vllm-runtime-template`. To create an instance of it, run the
+following command:
+```shell
+oc process -n opendatahub -o yaml vllm-runtime-template | oc apply -f -
+```
+
+You can verify that the `ServingRuntime` has been created successfully with the
+following command:
+```shell
+oc get servingruntimes
+
+NAME           DISABLED   MODELTYPE   CONTAINERS         AGE
+vllm-runtime              vLLM        kserve-container   11s
+```
+
+Notice that the ServingRuntime has been created with `vllm-runtime` name.
+
+Now that the `ServingRuntime` is configured, the Granite model stored in an OCI image can
+be deployed by creating an `InferenceService` resource:
+```yaml
+apiVersion: serving.kserve.io/v1beta1
+kind: InferenceService
+metadata:
+  name: sample-isvc-using-oci
+spec:
+  predictor:
+    model:
+      runtime: vllm-runtime # This is the name of the ServingRuntime resource
+      modelFormat:
+        name: vLLM
+      storageUri: oci://quay.io/<user_name>/<repository_name>:<tag_name>
+      args:
+        - --dtype=half
```

Comment:
is this parameter needed?
If so, we could add a small explanation about it

Reply:
I think it depends on the GPU and the model. At least in the Granite example, if the GPU has compute capability >= 8, it is not needed (vLLM logs made this clear). So, for this specific example it do is needed.

Reply:
Ok, can you please add a small quote about it and why is it needed?

Reply:
I added a `[!NOTE]`.